lib/msf/core/auxiliary/udp_scanner.rb

```diff
@@ -129,7 +127,8 @@ def scanner_send(data, ip, port)
     resend_count = 0
 
     begin
-      unless IPAddress(ip).multicast?
+      addrinfo = Addrinfo.ip(ip)
+      unless addrinfo.ipv4_multicast? || addrinfo.ipv6_multicast?
         sock = udp_socket(ip, port, bind_peer: true)
```

Comment:
As you can see, just using standard ruby is pretty trivial. Not sure if it's worth adding anything to rex (if anything, I wonder if we could convert some Rex Socket code to just be normal Ruby).